src/physics/base/ParticleMd.hh

```diff
@@ -0,0 +1,60 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file ParticleMd.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <string>
+#include "base/OpaqueId.hh"
+
+namespace celeritas
+{
+struct ParticleMd;
+using PDGNumber = OpaqueId<ParticleMd, int>;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Host-only metadata for a particle type.
+ *
+ * The PDG Monte Carlo number is a unique "standard model" identifier for a
+ * particle. See "Monte Carlo Particle Numbering Scheme" in the "Review of
+ * Particle Physics":
+ * http://pdg.lbl.gov/2019/reviews/rpp2019-rev-monte-carlo-numbering.pdf
+ *
+ */
+struct ParticleMd
+{
+    std::string name;     // Particle name
+    PDGNumber   pdg_code; // See "Review of Particle Physics"
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * \namespace pdg
+ *
+ * Unique standard model particle identifiers by the Particle Data Group.
+ *
+ * These should be extended as needed.
+ */
+namespace pdg
+{
```

Comment:
You got me. I guess the difference in my mind is "namespace for a collection of functions" vs "enum-like namespace", though that makes me think maybe this should be
```
enum class Pdg : int
{
    electron = 11,
    photon = 22,
    ...
};
```
and replace the opaque ID with that enum class.

Reply:
I guess the reason not to do this is that "PDG numbers" are technically more like encodings (like ZAIDs) than enumerations (like MTs). @pcanal care to weigh in when you're back from vacation?

Reply:
Technically it is not a closed/fixed set, albeit a very slowly evolving one ... I think that in Geant4 there is the possibility to add your own exotic particle species, for example.   

Most of the code (but not all) relies on the particle object being a 'singleton' per species for selection:
```
G4bool G4eBremsstrahlung::IsApplicable(const G4ParticleDefinition& p)
{
  return (&p == G4Electron::Electron() || &p == G4Positron::Positron());
}
```
(I am not commenting here whether this is a good or bad way to go, just describing).

In our case, since we support __less___ customizability, it might make sense to use an enum ... depending how much we end up relying on it may make it more or less hard to re-introduce the customizability later on.

Reply:
In GeantV, in addition to the PDGCode is supplemented by an internal code with a more compact range (literally the index in the array in Particle object, so usable for quick lookup) and the PDGCode is not really used internally.

Reply:
@pcanal The "internal code" sounds exactly like the `ParticleDefId`: the index in the list of `ParticleDef`s. The PDG codes for our purpose are a well-defined and unique identifier for looking up the particle def IDs. I envision PDG numbers as an intermediary for initializing Celeritas "process" classes with data exported from Geant4.

Reply:
Based on the discussion and the work with Stefano importing particles, I think these can stay in a `pdg` "namespace" that acts like an enum.